### ISSUE TITLE
Docs

### ISSUE THREAD

**Issue by FluorescentHallucinogen** [material-foundation/material-color-utilities#13]
The packages from this project definitely need docs. :)

**Comment by guidezpl** [comment]
Can you please elaborate? There's only a Dart package at the moment.

**Comment by FluorescentHallucinogen** [comment]
Yes, currently there's only a Dart package,but it seems TypeScript npm package [is on horizon](https://github.com/material-foundation/material-color-utilities/pull/10). :)

I mean docs for every function and how to use them. Maybe auto-generated from source code annotations. But samples could be very useful too.

**Comment by guidezpl** [comment]
I agree, complete documentation is on the roadmap! Stay tuned

**Comment by FluorescentHallucinogen** [comment]
One of the samples I'm very interested in is how to pass image to `material-color-utilities`. Obviously, some sort of [image selector](https://github.com/material-foundation/material-theme-builder/issues/15) is needed. But what next? What data format does it expect? Should I use Web Workers, WebAssembly, etc.

**Comment by FluorescentHallucinogen** [comment]
Just found cool demo for inspiration: https://lokeshdhakar.com/projects/color-thief/. :wink:

**Comment by guidezpl** [comment]
I'm currently working on a new module to add to the libraries, called `extract`. It'll be make the process really easy, give it some image and it returns top colors. Coming soon!

**Comment by selfire1** [comment]
Hi! I would love to use this project in TypeScript to generate a colour palette from a single source colour. Being in a bit over my head with how to use these utilities, having additional documentation would be great!

**Comment by rodydavis** [comment]
Typescript example added :)

**Comment by Galileo01** [comment]
The document is necessary
